src/AppInstallerCommonCore/Public/winget/ManifestYamlParser.h

```diff
@@ -3,61 +3,51 @@
+        std::string FileName;
 
-    private:
-        // These pointers are referenced in the processing functions in manifest field info table.
-        YAML::Node* m_p_installersNode = nullptr;
-        YAML::Node* m_p_switchesNode = nullptr;
-        YAML::Node* m_p_localizationsNode = nullptr;
-        AppInstaller::Manifest::Manifest* m_p_manifest = nullptr;
-        AppInstaller::Manifest::ManifestInstaller* m_p_installer = nullptr;
-        std::map<ManifestInstaller::InstallerSwitchType, Utility::NormalizedString>* m_p_switches = nullptr;
-        AppInstaller::Manifest::ManifestLocalization* m_p_localization = nullptr;
-
-        // This struct contains individual app manifest field info
-        struct ManifestFieldInfo
-        {
-            ManifestFieldInfo(std::string name, std::function<void(const YAML::Node&)> func, bool required = false, std::string regex = {}) :
-                Name(std::move(name)), ProcessFunc(func), Required(required), RegEx(std::move(regex)) {}
-
-            std::string Name;
-            std::function<void(const YAML::Node&)> ProcessFunc;
-            bool Required = false;
-            std::string RegEx = {};
-        };
-
-        std::vector<ManifestFieldInfo> RootFieldInfos;
-        std::vector<ManifestFieldInfo> InstallerFieldInfos;
-        std::vector<ManifestFieldInfo> SwitchesFieldInfos;
-        std::vector<ManifestFieldInfo> LocalizationFieldInfos;
-
-        std::vector<ValidationError> ParseManifest(const YAML::Node& rootNode, Manifest& manifest, bool fullValidation);
-
-        static std::map<ManifestInstaller::InstallerSwitchType, Utility::NormalizedString> GetDefaultKnownSwitches(
-            ManifestInstaller::InstallerTypeEnum installerType);
-
-        // This method takes YAML root node and list of manifest field info.
-        // Yaml-cpp does not support case insensitive search and it allows duplicate keys. If duplicate keys exist,
-        // the value is undefined. So in this method, we will iterate through the node map and process each individual
-        // pair ourselves. This also helps with generating aggregated error rather than throwing on first failure.
-        static std::vector<ValidationError> ValidateAndProcessFields(
-            const YAML::Node& rootNode,
-            const std::vector<ManifestFieldInfo>& fieldInfos,
-            bool fullValidation);
-
-        void PrepareManifestFieldInfos(const ManifestVer& manifestVer);
+        ManifestTypeEnum ManifestType;
     };
+
+    // fullValidation: Bool to set if manifest creation should perform extra validation that client does not need.
+    //                 e.g. Channel should be null. Client code does not need this check to work properly.
+    // throwOnWarning: Bool to indicate if an exception should be thrown with only warnings detected in the manifest.
+    // resourceDll:    Binary where schemas are embedded, or nullptr if they are embedded in the binary that created the process

```

Comment:
>// resourceDll:    Binary where schemas are embedded, or nullptr if they are embedded in the binary that created the process [](start = 4, length = 124)

This seems like it is only marginally useful for some negative test case.  Isn't it possible to determine the module that the current code is executing in?

Reply:
It's mainly for WinGetUtil.dll, I cannot find a simple way to detect the code is being called by wingetutil(or you prefer a bool like isFromWinGetUtil)?

---
In reply to: [571327922](https://github.com/microsoft/winget-cli/pull/740#discussion_r571327922) [](ancestors = 571327922)

Reply:
Either use GetModuleHandleEx or [https://devblogs.microsoft.com/oldnewthing/20041025-00](https://devblogs.microsoft.com/oldnewthing/20041025-00)

---
In reply to: [571330356](https://github.com/microsoft/winget-cli/pull/740#discussion_r571330356) [](ancestors = 571330356,571327922)